.github/workflows/integration-test.yml

```diff
@@ -15,6 +15,7 @@ jobs:
       fail-fast: false
       matrix:
         os: [ ubuntu-latest , windows-latest ]
+        datasource_version: [2.4.0, 2.13.0, 3.0.0]
```

Comment:
Will this check out the latest changes from main for 3.0.0 or will it get the latest published artifacts?

For unreleased code, we should aim to get the head of the main branch for core and the security plugin to catch issues early.

Reply:
This will download the min-distribution of opensearch (which should be stable), and then download the plugin zip from maven here: https://aws.oss.sonatype.org/content/repositories/snapshots/org/opensearch/plugin/opensearch-security/. There is a workflow in the security backend plugin to publish a zip on each merge into the main/2.x/1.x lines, so it should be the latest from the security backend plugin.

Reply:
How is this being tested elsewhere?

Reply:
Currently, I don't think other plugins and dashboards are testing this backwards compatibility feature - it is something we are adding for our own repo. I added the 1.3 backport of the maven publish workflow here: https://github.com/opensearch-project/security/pull/4285. Once that finishes, we should be able to add 1.3.16/1.3.17 into this matrix as well to test, but I don't think that should be a blocker for this change to go in. Resolving this conversation for now.

Reply:
@kkhatua Can you help us get an owner for this.  IMO this is outside the scope of efforts as a plugin team.

Reply:
I'm OK with this as long as we are getting better testing quality. Other plugins can follow suite or deal with production issues later on.